priv/filelock.go

```diff
@@ -20,70 +20,40 @@
 package priv
 
 import (
-	"errors"
-	"os"
-	"syscall"
+	sys "syscall" // XXX: if we are only targeting 15.04+, i.e. golang 1.5+, we should move to golang.org/x/sys
 )
 
-// ErrNotLocked is returned when an attempts is made to unlock an
-// unlocked FileLock.
-var ErrNotLocked = errors.New("not locked")
-
-// FileLock is a Lock file object used to serialise access for
-// privileged operations.
-type FileLock struct {
-	Filename string
-	realFile *os.File
-}
-
-// NewFileLock creates a new lock object (but does not lock it).
-func NewFileLock(path string) *FileLock {
-	return &FileLock{Filename: path}
-}
-
-// Lock the FileLock object.
-// Returns ErrAlreadyLocked if an existing lock is in place.
-func (l *FileLock) Lock(blocking bool) error {
-	// XXX: don't try to create exclusively - we care if the file failed to
-	// be created, but we don't care if it already existed as the lock
-	// _on_ the file is the most important thing.
-	flags := (os.O_CREATE | os.O_WRONLY)
-
-	f, err := os.OpenFile(l.Filename, flags, 0600)
+// LockedFile is a handle you can use to Unlock a file lock.
+type LockedFile int
+
+// FileLock opens (and possibly creates) a new file at the given path
+// and applies an exclusive advisory lock on it.
+//
+// If the file already has an advisory lock, the `blocking` flag
+// determins whether to block until it is removed or return
+// ErrAlreadyLocked.
+func FileLock(path string, blocking bool) (LockedFile, error) {
```

Comment:
It's very common to have verbs as bool parameters. Grep some arbitrary code and you'll find a few.

The distinction, both for the function name and for the parameter name, is whether this is doing the action right then, or setting up for a later action. FileLock is a noun, LockFile is an action. Blocking is a property for future use, but block (or "wait") is a request to do that right now.

In either case, these may be a non-issue if you agree to do the refactoring to turn both of these APIs into the single one suggested.